src/screens/Ticketing/Purchase/Overview/index.tsx

```diff
@@ -101,7 +108,15 @@ const PurchaseOverview: React.FC<OverviewProps> = ({
             />
           )}
 
-          {durationSelectionEnabled && (
+          {productSelectionMode === 'product' && (
+            <ProductSelection
+              selectedProduct={preassignedFareProduct}
+              setSelectedProduct={onSelectPreassignedFareProduct}
+              style={styles.selectionComponent}
+            />
+          )}
+
+          {productSelectionMode === 'duration' && (
```

Comment:
I think it would be cleaner if this was wrapped in a `ProductSelection`-component, which then decides the component based on the product selection mode.